vj4/app.py

```diff
@@ -33,13 +34,43 @@
                help='Expire time for changemail token, in seconds.')
 options.define('url_prefix', default='https://vijos.org', help='URL prefix.')
 options.define('cdn_prefix', default='/', help='CDN prefix.')
+options.define('sentry_dsn', default='', help='Sentry integration DSN.')
 
 _logger = logging.getLogger(__name__)
 
 
+class SentryMiddleware(aiohttp_sentry.SentryMiddleware): # For getting a correct IP
+  async def get_extra_data(self, request):
+    return {
+      'request': {
+        'query_string': request.query_string,
+        'cookies': request.headers.get('Cookie', ''),
+        'headers': dict(request.headers),
```

Comment:
`dict()` is expensive, consider `list()`

Reply:
Must be dict, or raven will crash